backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,419 @@
+        Assert.assertEquals(taskViewModel.updateCancelEditInLayout.value, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, false)
+    }
+
+    @Test
+    fun testNext_ShouldUpdateCurrentStepToNextStep_shouldUpdateCurrentStepEventToNextStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, nextStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, nextStepMocked)
+    }
+
+    @Test
+    fun testPrevious_ShouldUpdateCurrentStepToPreviousStep_shouldUpdateCurrentStepEventToPreviousStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+
+        val currentStepMocked = createCurrentStep()
+        val previousStepMocked = createCurrentStep("previousStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(previousStepMocked)
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, previousStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, previousStepMocked)
+    }
+
+    @Test
+    fun checkPrevious_shouldCloseWhenPreviousIsNull() {
+        // Assemble
+        taskViewModel = createViewModel()
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(null)
+        val currentStepMocked = createCurrentStep()
+        taskViewModel.currentStep = currentStepMocked
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, false)
+    }
+
+    @Test
+    fun testNext_shouldCloseWhenNextNull() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, true)
+    }
+
+    @Test
+    fun testShowCurrentStep_ifCurrentStepIsNull_ShowGoToNextStep() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        // Act
+        taskViewModel.showCurrentStep()
+
+        // Assert
+        verify(taskViewModel, times(1)).showCurrentStep()
+    }
+
+    @Test
+    fun testPreviousOnEditMode_shouldCheckForCancelDialog() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        val currentStepMocked = createCurrentStep()
+        taskViewModel.edit(currentStepMocked)
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        verify(taskViewModel, times(1)).showCancelEditAlert()
```

Comment:
Like Above, this is not truly testing the behavior of the method, just that it got called. I haven't looked at all the tests yet (I'm going 1 by 1) so If there's one below that tests for this, disregard this comment.

Reply:
this test is not necessarily, going to remove it